--- a/libavcodec/aac_ac3_parser.c
+++ b/libavcodec/aac_ac3_parser.c
@@ -143,6 +143,8 @@
             if (hdr.bitstream_mode == 0x7 && hdr.channels > 1)
                 avctx->audio_service_type = AV_AUDIO_SERVICE_TYPE_KARAOKE;
             bit_rate = hdr.bit_rate;
+#else
+            return i; /* No parser is compiled for this codec. */
 #endif
         } else {
 #if CONFIG_AAC_PARSER
@@ -157,6 +159,8 @@
             /* ADTS does not support USAC */
             s1->key_frame = 1;
             bit_rate = hdr.bit_rate;
+#else
+            return i; /* No parser is compiled for this codec. */
 #endif
         }
 
--- a/libavcodec/h263dec.c
+++ b/libavcodec/h263dec.c
@@ -612,7 +612,9 @@
             ff_msmpeg4_decode_ext_header(h, buf_size) < 0)
             h->c.er.error_status_table[h->c.mb_num - 1] = ER_MB_ERROR;
 
+#if CONFIG_WMV2_DECODER
 frame_end:
+#endif
     if (!h->c.studio_profile)
         ff_er_frame_end(&h->c.er, NULL);
 
--- a/libavformat/rawdec.c
+++ b/libavformat/rawdec.c
@@ -103,6 +103,8 @@
     return 0;
 }
 
+
+#if CONFIG_DATA_DEMUXER
 static int raw_data_read_header(AVFormatContext *s)
 {
     AVStream *st = avformat_new_stream(s, NULL);
@@ -113,6 +115,8 @@
     st->start_time = 0;
     return 0;
 }
+
+#endif
 
 /* Note: Do not forget to add new entries to the Makefile as well. */
 
